app/components/primer/beta/auto_complete.rb

```diff
@@ -32,14 +32,45 @@ class AutoComplete < Primer::Component
         Primer::BaseComponent.new(**system_arguments)
       }
 
+      # Customizable input used to search for results.
+      # It is preferred to use this slot sparingly - it will be created by default if not explicity added.
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :input, lambda { |**system_arguments|
+        sanitized_args = deny_tag_argument(**system_arguments)
+        sanitized_args = deny_single_argument(:autofocus, "autofocus is not allowed for accessibility reasons. See https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus#accessibility_considerations for more information.", **sanitized_args)
+        deny_aria_key(
+          :label,
+          "instead of `aria-label`, include `label_text` and set `is_label_visible` to `false` on the component initializer.",
+          **sanitized_args
+        )
+        sanitized_args[:id] = @input_id
+        sanitized_args[:name] = @input_id
```

Comment:
Should we also explicitly deny `id` and `name` given that it is ignored and overridden by `@input_id`?

Reply:
Ooh that is a good idea - I say we go with your suggestion and revisit if needed!

Reply:
Ok, I added an explicit deny for `id` and `name` in https://github.com/primer/view_components/pull/1050/commits/7fd497d24fc6623f2384e586fc2c5d73ca8a9e19 and tests in https://github.com/primer/view_components/pull/1050/commits/d2e7cfcad7a16d4a1096c3645f58eff104aaeeef. 

On second thought, I feel like we should allow `name` to be overridden since `name` and `id` have different functions. Down to revisit this later though!

Reply:
I've added the `input_name` option back in 🙂